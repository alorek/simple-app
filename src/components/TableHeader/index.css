th.sorting-header {
    cursor: pointer;
}
