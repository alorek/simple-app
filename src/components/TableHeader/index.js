import React, { Component } from 'react';
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';

import { sortTable } from '../../actions';

class TableHeader extends Component {

    getSortIcon() {

        const sortOrder = this.props.sortOrder;
        const triangleDirection = sortOrder === 'asc' ? 'top' : 'bottom';
        return `pull-right glyphicon glyphicon-triangle-${triangleDirection}`;
    }

    getNewSortOrder(columnId) {

        if (columnId === this.props.sortKey) {
            return this.props.sortOrder === 'asc' ? 'desc' : 'asc';
        }
        return 'asc';
    }

    render() {
        const column = this.props.column;

        const showSorting = column.id === this.props.sortKey;

        return (
            <th
                className="sorting-header"
                onClick={ () => { this.props.sortTable(column.id, this.getNewSortOrder(column.id)) } }>

                <span className="pull-left">{ column.label }</span>

                { showSorting ?
                    <span className={ this.getSortIcon() }></span> :
                    null
                }
            </th>

        );
    }
}

function mapDispatchToProps (dispatch) {
    return bindActionCreators({ sortTable: sortTable }, dispatch);
}

function mapStateToProps(state) {
    return {
        sortKey: state.table.sortKey,
        sortOrder: state.table.sortOrder
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(TableHeader);
