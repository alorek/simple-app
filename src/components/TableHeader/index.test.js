import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import TableHeader from './index';
import reducer from '../../reducers';

const store = createStore(reducer);

test('TableHeader renders correctly', () => {
    const column = {id: 'resource_type'};

    const tree = renderer.create(
        <Provider store={ store }>
            <Router>
                <TableHeader column={ column }/>
            </Router>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
