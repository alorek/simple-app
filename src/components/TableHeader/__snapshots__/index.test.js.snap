// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader renders correctly 1`] = `
<th
  className="sorting-header"
  onClick={[Function]}
>
  <span
    className="pull-left"
  />
  <span
    className="pull-right glyphicon glyphicon-triangle-top"
  />
</th>
`;
